src/main/java/DecoratorForString.java

```diff
@@ -0,0 +1,46 @@
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class DecoratorForString implements Links{
+
+    protected String url;
+
+    public DecoratorForString(String url) {
+        this.url = url;
+    }
+
+    private List<String> prepareLinks() {
+
+        List<String> links  = new ArrayList<String>();
+        Document doc;
+        try {
+            doc = Jsoup.connect(url).get();
+            Elements elements = doc.body().getElementsByTag("a");
+
+            for (Element element : elements) {
+                String urlName =  element.attr("abs:href");
+                if (!urlName.equals("")) {
+                    links.add(urlName);
+                }
+            }
+
+        } catch (IllegalStateException e) {
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
@PavelNejinskiy not decide what to do with this exception(catching and printing is considered as particular decision).
Instead I'd suggest to just re-throw as `IllegalStateException`, since we do not really care about error handling just now